### PR TITLE
feat(plugin): add configurable `plugin`

### PR DESCRIPTION
**Pull request by guillei10**
The goal of this PR is to provide a set of function options to configurate a tailwind plugin. We have created some test styles to be removed once we have some designs for the base styles.

- [ ] Dependencies. If any, specify:
- [x] Open issue. If applicable, link: [EX-5701](https://searchbroker.atlassian.net/browse/EX-5701)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

You can test it by running `npm run dev` once the build is done and play with the `index.html` and the `options.ts`

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/tailwindcss-x/.eslintrc.js [hunk 1]
```diff
@@ -2,5 +2,14 @@ module.exports = {
...
     tsconfigRootDir: __dirname,
     project: 'tsconfig.eslint.json'
-  }
+  },
+  overrides: [
+    {
+      files: ['src/**/*.ts'],
+      rules: {
+        '@typescript-eslint/no-unsafe-call': 'off',
+        '@typescript-eslint/no-unused-vars': 'off'
```
Comment (LuisMartinez15) [inline packages/tailwindcss-x/.eslintrc.js:11]: I remember putting this rule at the beginning, but it seems it is no longer necessary and can be safely removed now.

File: packages/tailwindcss-x/src/tailwind-x.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import plugin from 'tailwindcss/plugin';
+import { deepMerge } from '@empathyco/x-deep-merge';
+import { forEach } from '../../x-utils';
```
Comment (LuisMartinez15) [inline packages/tailwindcss-x/src/tailwind-x.plugin.ts:3]:
```suggestion
import { forEach } from '@empathyco/x-utils';
```

File: packages/tailwindcss-x/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { Dictionary } from '@empathyco/x-utils';
+import { TailwindPluginFn } from 'tailwindcss/plugin';
+import { AnyFunction } from '../../x-utils';
```
Comment (LuisMartinez15) [inline packages/tailwindcss-x/src/types.ts:3]:
```suggestion
import { AnyFunction, Dictionary } from '@empathyco/x-utils';
import { TailwindPluginFn } from 'tailwindcss/plugin';
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
Good job. The plugin is taking shape! 🚀

File: packages/tailwindcss-x/demo/options.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    return {
+      '.icon': {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        width: theme('spacing.16'),
+        height: theme('spacing.16'),
+        color: 'currentColor',
+        fontSize: theme('spacing.8')
+      }
```
Comment (tajespasarela) [inline packages/tailwindcss-x/demo/options.ts:14]: The types are not working. Here we should have autocomplete for overriding default components and is not working.

We should have this:
image

The same for utilities and dynamic components/utilities

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/tailwindcss-x/demo/options.ts (hunk 4 above)
Comment (LuisMartinez15) [reply packages/tailwindcss-x/demo/options.ts:20]: Fixed

File: packages/tailwindcss-x/demo/vite.config.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import autoprefixer from 'autoprefixer';
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/tailwindcss-x/demo/vite.config.js:1]: I prefer to have all code inside `src` and leave out the config files. It is strange to me to have this code outside. But as you prefer.

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**